File: python/101-150/102. Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        res = []
        if not root:
            return res
        def helper(root):
            queue = [root]
            while queue:
                temp = []
                level = []
                for node in queue:
                    level += [node.val]
                    if node.left:
                        temp.append(node.left)
                    if node.right:
                        temp.append(node.right)
                res.append(level)
                queue = temp
        helper(root)
        return res

File: cpp/101-150/102. Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
	vector<vector<int>> levelOrder(TreeNode* root) {
		vector<vector<int>> res;
		if(!root)
			return res;
		vector<TreeNode *> level;
		level.push_back(root);
		while(level.size() != 0){
			vector<TreeNode *> temp;
			vector<int> level_vals;
			for(auto node : level){
				level_vals.push_back(node->val);
				if(node->left)
					temp.push_back(node->left);
				if(node->right)
					temp.push_back(node->right);
			}
			res.push_back(level_vals);
			level = temp;
		}
		return res;
	}
};